client/js/shout.js

```diff
@@ -197,6 +197,23 @@ $(function() {
 				data.msg
 			]);
 
+		chan.find(".messages .text:not(.processed)")
+			.each(function() {
+				var html = $(this).html();
+				$(this)
+					.addClass("processed")
+					.html(html.replace(/(\W)(#[\w\d-.]{0,200})/ig,
```

Comment:
Thanks for linking to an RFC :clap: 
Actually, it's so useful to have it there, could you make it a comment shaped like this please:

```js
					// Channels names are strings (beginning with a '&' or '#' character)
					// of length up to 200 characters.
					// See https://tools.ietf.org/html/rfc1459#section-1.3
					.html(...
```

Three other comments:
- I don't see support for `&channel`, as only `#` is present in your regex. Could you add support for this please?
- Even though the RFC doesn't explicitly say it, I don't think  `#` is a legal channel. So maybe minimal length should be `1` instead of `0`
- The RFC says, however, that channel names are up to 200 characters, `#`/`&` included :-) So it should be `199` and not `200` :-)